core/src/main/java/org/opensearch/sql/analysis/QualifiedNameResolver.java

```diff
@@ -0,0 +1,78 @@
+/*
+ *
+ *  * Copyright OpenSearch Contributors
+ *  * SPDX-License-Identifier: Apache-2.0
+ *
+ */
+
+package org.opensearch.sql.analysis;
+
+import java.util.List;
+import java.util.Set;
+import lombok.experimental.UtilityClass;
+import org.apache.commons.lang3.StringUtils;
+import org.opensearch.sql.ast.expression.QualifiedName;
+import org.opensearch.sql.catalog.model.CatalogSchemaTableName;
+import org.opensearch.sql.catalog.model.SchemaTableName;
+
+@UtilityClass
+public class QualifiedNameResolver {
+
+  public static final String DEFAULT_CATALOG_NAME = ".opensearch";
+  public static final String DEFAULT_SCHEMA_NAME = "default";
+  public static final String INFORMATION_SCHEMA_NAME = "information_schema";
+  private static final String DOT = ".";
+
+  /**
+   * Resolve Given QualifiedTableName to CatalogSchemaTableName.
+   * @param name name.
+   * @param availableCatalogs availableCatalogs.
+   * @return CatalogSchemaTableName catalogSchemaTableName.
+   */
+  public static CatalogSchemaTableName resolveTableName(QualifiedName name,
```

Comment:
Does the logic apply to all the identifier or just for table name?

Reply:
https://github.com/opensearch-project/sql/issues/896